### ISSUE TITLE
Broken checks in TagsCheck.py

### ISSUE THREAD

**Issue by thisisshub** [rpm-software-management/rpmlint#454]
These tags encountered in the `TagsCheck.py` are either partially broken or totally broken
**Disclaimer:**
- [ ] Broken
- [x] Fixed 

- Partially broken checks
 - [x]  summary-on-multiple-lines
	 
   - **Definition:**
	    
        Your summary must fit on one line.     
	
		-  **Where is it found?**
    
			This warning should be encountered in packages that have a multi line summary. But you can still not find the warning in the stdout.

     - **How to reproduce this error?**
         - Make a simple spec file with multiple line summary, using vim you can type in insert mode ```Ctrl-v + <Enter Key> or CR``` which would show a highlighted `^M` character (which acts like a enter key).
        			            
         - or try making a `Summary: This is a \n summary` tag line since the code suggests in the `check_summary()` method 
         
          `if '\n' in summary:
              self.output.add_info('E', pkg, 'summary-on-multiple-lines', lang)`
              
         
         - `rpmbuild -bb <or -bs> package.spec`
			   
         - run `rpmlint package.rpm` 
         
   - **Expected Result**:
		
         package.x86_64: W: summary-on-multiple-lines rpmbuild/RPMS/x86_64 package-0-1.x86_64.rpm  
	
	-	**Reality**:
		
          -	No warning in stdout 

 - [x]  invalid-license
 
	 - **Definition:**

         (definition not available in the  TagsCheck.toml. Below is a user defined definition)
	       The license should be valid license which conforms to the SPDX shortname format
		
    -  **Where is it found?**
			
      This warning should be encountered in packages that have a invalid license. But you can still find the warning in the stdout of every valid license containing package.
	   
     - **How to reproduce this error?**
			   
         - Make a simple spec file with `License: MIT` or any other valid license
			   
         - `rpmbuild -bb <or -bs> package.spec`
			   
         -  run `rpmlint package.rpm` 
   
   - **Expected Result**:
		   
       -   No warning in stdout
	
	- **Reality**: 
			
	   -  `package.x86_64: W: invalid-license MIT`

 - [ ]  tag-not-utf8
	 
   - **Definition:**
	       
         The character encoding of the value of this tag is not UTF-8.
	
		-  **Where is it found?**
			
        This warning should be encountered in packages that have a valid license. But you cannot find the warning in the stdout of the concerned package.
	
	   - **How to reproduce this error?**
			
      - For example in vim, goto insert mode and press `Ctrl-v` and type `0x00` which would make `^@` null sign since, `NUL == 0x00 == 0 == Ctrl + @ == ^@` (more info [here]([https://stackoverflow.com/questions/71323/how-to-replace-a-character-by-a-newline-in-vim](https://stackoverflow.com/questions/71323/how-to-replace-a-character-by-a-newline-in-vim))) in vim. Type this character in any tag value. Hence,  
	
 Make a simple spec file with say, 
				
```
#1
Summary: Something <non-utf8-character> warning
	    
#2
%description
This is a summary with <non-utf8-character>

#3
%changelog
Someone <non-utf8-character> Something    
```

   - **Reality:**

		- `/path/to/package.spec does not appear to be a specfile.`
		
		- No warning in stdout

	  
   - **Expected Result**:

		- `package.x86_64: E: tag-not-utf8 Summary Something <non-utf8-character> warning `

		- `package.x86_64: E: tag-not-utf8 description This is a summary with <non-utf8-character>`

		- `package.x86_64: E: tag-not-utf8 changelog Someone <non-utf8-character> Something`

**Comment by thisisshub** [comment]
**Edit 1:**

`invalid-license` check has been correctly tested by usage of `configdefaults.toml`,

```python
@pytest.mark.parametrize('package', ['binary/invalid-license'])
def test_check_invalid_license(tmpdir, package, tagscheck):
    """Test if a package check,
    - in out,
        invalid-license
    - not in out,
        invalid-license."""
    CONFIG.info = True
    CONFIG.configuration['ValidLicenses'] = ['MIT']
    CONFIG.configuration['ReleaseExtension'] = "^0"
    output = Filter(CONFIG)
    test = TagsCheck(CONFIG, output)
    test.check(get_tested_package(package, tmpdir))
    out = output.print_results(output.results)
    # Test if a package has a License: value apart from ValidLicenses = []
    # list in configdefaults in its specfile
    assert 'W: invalid-license Apache License' in out
    # Test if a package has a License: value from ValidLicenses = []
    # ValidLicenses = ['MIT'] in this case in configdefaults in its specfile
    assert 'W: invalid-license MIT' not in out
```

**Comment by marxin** [comment]
> [ ] summary-on-multiple-lines

Are you sure one can really create a spec file that will really contain a line break? `\n` is not a line-break for RPM, it's backslash and `n`.

**Comment by thisisshub** [comment]
@marxin,
I mean that's just one of the things that I tried, rather it was one of the things on top of my mind to achieve a newline in specfile.  I mean a literal "return character" (enter key), a `\n` or a `^M` character were the things that I tried but none achieved a line break.

**Comment by marxin** [comment]
> I mean that's just one of the things that I tried, rather it was one of the things on top of my mind to achieve a newline in specfile. I mean a literal "return character" (enter key), a `\n` or a `^M` character were the things that I tried but none achieved a line break.

Sure, but I don't see how is the `"summary-on-multiple-lines"` broken. It's likely not possible (or very hard) to trigger it.

**Comment by thisisshub** [comment]
@marxin,
Well, technically it is broken.
Since, if you look at the code in `checks/TagsCheck.py`

```python3
if '\n' in summary:
    self.output.add_info('E', pkg, 'summary-on-multiple-lines', lang)
```

`\n` is supposed to be the character that triggers the error. But I have even tried `return` key and `^M` character which do not work either.

**Comment by marxin** [comment]
> `\n` is supposed to be the character that triggers the error. But I have even tried `return` key and `^M` character which do not work either.

All right, I've just made a PR that tests that.

**Comment by marxin** [comment]
> [ ] tag-not-utf8
> 
> *

About this, I found it not possible to build such a RPM package as `rpmbuild` rejects that:

```
$ rpmbuild *spec
error: bad date in %changelog: invalid3�
error: Package invalid-utf8-characters: invalid utf-8 encoding in Summary: invalid� - Invalid or incomplete multibyte or wide character
error: Package invalid-utf8-characters: invalid utf-8 encoding in Description: invalid2� - Invalid or incomplete multibyte or wide character

```
